File: src/model/Transaction.java
package model;

import model.TransactionCategories.TransactionCategory;

import java.util.Date;

public class Transaction {

	//Fields
	private double amount;
	private String party;	//name of the store, person, etc. that the transaction is going to/came from.
	private Date date;
	private TransactionCategory category;

	//Methods
	public void DetermineCategory() {
		//based on the party String
	}

	//Getters and Setters
	public double getAmount() {
		return amount;
	}

	public void setAmount(double amount) {
		this.amount = amount;
	}

	public String getParty() {
		return party;
	}

	public void setParty(String party) {
		this.party = party;
	}

	public Date getDate() {
		return date;
	}

	public void setDate(Date date) {
		this.date = date;
	}
}

File: FinanceTracker/FinanceTracker/Model/Transaction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FinanceTracker.Model {
	class Transaction {

		//Fields
		private string participant;
		private DateTime date;
		private double amount;
		private string type;

		//Constructor
		public Transaction(string participant, DateTime date, double amount) {
			this.participant = participant;
			this.date = date;
			this.amount = amount;
		}

		//Getters
		public string getParticipant() {
			return participant;
		}

		public DateTime getDate() {
			return date;
		}

		public double getAmount() {
			return amount;
		}
		
		public string getType() {
			return type;
		}
	}
}
